### ISSUE TITLE
Keyboard navigation of RadioButton groups doesn't allow selection

### ISSUE THREAD

**Issue by caltemose** [grommet/grommet#2676]
Navigating a group of RadioButton instances via keyboard only allows you to select the first radio button; subsequent buttons in the group cannot be selected. Additionally, the focus states can cause confusion as you navigate forward and backward through the buttons as the focus states either don't appear or get stuck on when they should be off.

### Expected Behavior

Users should be able to navigate through a group of radio buttons using the same keyboard controls as they'd use with standard HTML inputs of type radio. Normally a user will:

1. Use 'tab' to enter a radio button group and the first radio button is focused.
2. Use the arrow keys to navigate within the group to select other radio buttons.
3. Use space to select their desired radio button.
4. Use tab to leave the radio button group.

(I may have left out details - ultimately this should navigate via keyboard exactly as a basic HTML set of radio inputs.)

### Actual Behavior

1. Tab into the radio group; the first radio button is focused.
2. Type spacebar, radio button does not appear selected.
3. Use tab to navigate to next button; it does not receive a visible focus state.
4. Use tab again; no visible reaction.

Then to see additional odd behavior...

5. Make sure you've tabbed enough to leave the radio group (for example, using the code sandbox you could add a submit button after the radio button group).
6. Shift-tab to go back to the radios. The final radio button shows a focus state.
7. Keep shift-tabbing to the beginning of the radios; all radio buttons will show a focus state.

With enough playing around I realized you can do this sequence to get things to work - I don't mention this as a potential solution; only to help with debugging.

1. Tab into the radio button group.
2. Tab to the 2nd button. Hit the right arrow. The 2nd button will show as selected.
3. Tab to the 3rd button. Hit the right arrow. The 3rd button will show as selected.
etc.

### URL, screen shot, or Codepen exhibiting the issue

[https://codesandbox.io/s/6w4v63jjoz?module=%2Fsrc%2FRadioButton.js](https://codesandbox.io/s/6w4v63jjoz?module=%2Fsrc%2FRadioButton.js)

(You may need to select the RadioButton from the preview index)

### Steps to Reproduce

See actual behavior above. Note that when I preview the code sandbox version here the keyboard behavior is still weird but slightly different: 
[https://6w4v63jjoz.codesandbox.io/radiobutton](https://6w4v63jjoz.codesandbox.io/radiobutton)

### Your Environment

- Grommet version: 2.3.1 (in local version)
- Browser Name and version: Chrome 71.0.3578.98, Firefox 64/Quantum
- Operating System and version (desktop or mobile): Mac 10.12.6

**Comment by ShimiSun** [comment]
@petercpwong is your recent PR of focus fixes addressing the problem here?

**Comment by petercpwong** [comment]
@ShimiSun No my recent PR focused more on the visual focus indicators showing up as intended. It doesn't solve the issue OP raised here.
